store/sqlstore/channel_store.go

```diff
@@ -2302,6 +2302,104 @@ func (s SqlChannelStore) performSearch(searchQuery string, term string, paramete
 	return result
 }
 
+func (s SqlChannelStore) getSearchGroupChannelsQuery(userId, term string, isPostgreSQL bool) (string, map[string]interface{}) {
+	var query, baseLikeClause string
+	if isPostgreSQL {
+		baseLikeClause = "ARRAY_TO_STRING(ARRAY_AGG(u.Username), ', ') LIKE %s"
+		query = `
+            SELECT
+                *
+            FROM
+                Channels
+            WHERE
+                Id IN (
+                    SELECT
+                        cc.Id
+                    FROM (
+                        SELECT
+                            c.Id
+                        FROM
+                            Channels c
+                        JOIN
+                            ChannelMembers cm on c.Id = cm.ChannelId
+                        JOIN
+                            Users u on u.Id = cm.UserId
+                        WHERE
+                            c.Type = 'G'
+                        AND
+                            u.Username = (SELECT Username FROM Users WHERE Id = :UserId)
+                        GROUP BY
+                            c.Id
+                    ) cc
+                    JOIN
+                        ChannelMembers cm on cc.Id = cm.ChannelId
+                    JOIN
+                        Users u on u.Id = cm.UserId
+                    GROUP BY
+                        cc.Id
+                    HAVING
+                        %s
+                    LIMIT
+                        ` + strconv.Itoa(model.CHANNEL_SEARCH_DEFAULT_LIMIT) + `
+                )`
+	} else {
+		baseLikeClause = "GROUP_CONCAT(u.Username SEPARATOR ', ') LIKE %s"
+		query = `
+            SELECT
+                cc.*
+            FROM (
+                SELECT
+                    c.*
+                FROM
+                    Channels c
+                JOIN
+                    ChannelMembers cm on c.Id = cm.ChannelId
+                JOIN
+                    Users u on u.Id = cm.UserId
+                WHERE
+                    c.Type = 'G'
+                AND
+                    u.Username = (SELECT Username FROM Users WHERE Id = :UserId)
```

Comment:
Same here, I don't see the point of the Users join in this subquery, and the subquery to get the Username based on the :UserId.